server/src/main/java/org/spine3/server/projection/ProjectionRepository.java

```diff
@@ -287,21 +323,27 @@ public void setOnline() {
         setStatus(Status.ONLINE);
     }
 
-   /**
+    /**
      * The stream observer which redirects events from {@code EventStore} to
      * the associated {@code ProjectionRepository}.
      */
     private static class EventStreamObserver implements StreamObserver<Event> {
 
         private final ProjectionRepository projectionRepository;
+        private BulkWriteOperation operation;
 
         EventStreamObserver(ProjectionRepository projectionRepository) {
             this.projectionRepository = projectionRepository;
         }
 
         @Override
         public void onNext(Event event) {
+            if (operation == null || operation.isInProgress()) {
+                // Set the max time to complete the bulk to 40 seconds.
+                operation = projectionRepository.startBulkWrite(Durations.ofSeconds(40));
```

Comment:
@armiol I think we need to extract this into a framework parameter, e.g. in the gradle script.